webplugin/js/app/kommunicate-ui.js

```diff
@@ -404,14 +404,46 @@ setAvailabilityStatus : function (status){
 showClosedConversationBanner  : function(isConversationClosed){
     var messageText = MCK_LABELS["closed.conversation.message"];
     var conversationStatusDiv = document.getElementById("mck-conversation-status-box");
-    if (isConversationClosed){
+    var isCSATenabled = kommunicate._globals.collectFeedback;
+    var FEEDBACK_UPDATE_URL = "/feedback"
+    isConversationClosed && kommunicateCommons.modifyClassList( {class : ["mck-box-form"]}, "n-vis");
+    if(isCSATenabled && isConversationClosed && !kommunicateCommons.isConversationClosedByBot()){
+        mckUtils.ajax({
+            type: 'GET',
+            url: Kommunicate.getBaseUrl() + FEEDBACK_UPDATE_URL + '/'+ CURRENT_GROUP_DATA.tabId ,
+            global: false,
+            contentType: 'application/json',
+            success: function (data) {
+                CURRENT_GROUP_DATA.currentGroupFeedback = data.data;
+                kommunicateCommons.modifyClassList( {class : ["mck-box-form"]}, "n-vis");
+                kommunicateCommons.modifyClassList( {class : ["mck-csat-text-1"]}, "","n-vis");
+                kommunicateCommons.modifyClassList( {id : ["mck-sidebox-ft"]}, "mck-closed-conv-banner");
+                if( data.data && data.data.rating && data.data.comments.length> 0){
+                    kommunicateCommons.modifyClassList( {id : ["csat-3","mck-rated"]}, "", "n-vis");
+                    kommunicateCommons.modifyClassList( {id : ["csat-1","csat-2"]}, "n-vis");
+                    document.getElementById('csat-3').innerHTML = '\"' + data.data.comments[0] + '\"';
+                }else if( data.data && data.data.rating) {
+                    kommunicateCommons.modifyClassList( {id : ["csat-2","mck-rated"]}, "", "n-vis");
+                    kommunicateCommons.modifyClassList( {id : ["csat-1","csat-3"]}, "n-vis");
+                    document.getElementById('mck-rating-container').innerHTML = kommunicateCommons.getRatingSmilies(data.data.rating);
+                }else {
+                    kommunicateCommons.modifyClassList( {id : ["csat-1"]}, "", "n-vis");
```

Comment:
@akshatdhabi something looks wrong here, what if there are 5 type of ratings in future or lets say rating scale from 0 to 9, then do we need to write code for all 9 entries csat-2, csat-3, cast-4.... csat-9?

Reply:
These are not entries for the ratings, csat-1,2,3 are the containers of step 1,2,3.

Reply:
one easier way can be:
remove n-vis from all before starting if else and then inside if else conditions, only modify that respective container.

Reply:
The checks are there as the review can also be in parts, say ifa user has only rated the conversation but not commented on it we are supposed to show the comment box instead of final rating screen

Reply:
@devashishmamgain  done